lib/ast/expression.rb

```diff
@@ -46,6 +47,12 @@ def get_parameters(func_params)
     output << @parameters.first.code(func_params)
   end
 
+  def throw_return_error(expected_return_type)
+    raise CompileError, <<~ERROR
+      function ':#{@function}' returns #{FunctionDictionary[@function].values.first}, not #{expected_return_type}
+    ERROR
+  end
+
   def throw_compile_error
```

Comment:
rename to parameter error
```suggestion
  def throw_parameter_error
```